helix-term/src/commands/typed.rs

```diff
@@ -2167,6 +2169,64 @@ fn reset_diff_change(
     Ok(())
 }
 
+fn rename_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+    ensure!(args.len() == 1, ":rename takes one argument");
+    let new_name = args.first().unwrap();
+
+    let (_, doc) = current!(cx.editor);
+    let path = doc
+        .relative_path()
+        .ok_or_else(|| anyhow!("Buffer not saved, use :write"))?
+        .canonicalize()
+        .map_err(|_| anyhow!("Could not get absolute path of the document"))?;
+    let mut path_new = path.clone();
+    path_new.set_file_name(OsStr::new(new_name.as_ref()));
```

Comment:
note that this means that the destination directory might not exist so we need `rename!` too

Reply:
This is a sort of weird one, if we change the command name to `move` as previously said, probably a good ide, thoguh `mv` in bash takes absolute paths, not ones relative to the file but I think it's work

On the other hand I don't think a `file` renaming function should be able to move it as most file managers just let you rename it

Imo, `move` is probably better

Reply:
I guess it might make mose sense to just call the entire command move and make the destination a path relative to CWD. If somebody wants to just change the filename they can use `%filname/../<new_name>` in the future.

I think that generally move is a better idea than rename because it can do more (and the LS can handle full moves  for example RA can move a module to a totally different module